### PR TITLE
CSSRE-2260-Add-permissions-to-RHOC-clusters-for-CSSRE

### PR DESCRIPTION
**Pull request by pat-cremin**
### What type of PR is this?
feature

### What this PR does / why we need it?
The PR is to facilitate access to RHOC clusters and namespaces for the CSSRE team. 

### Which Jira/Github issue(s) this PR fixes?

_Fixes #_
[CSSRE-2260](https://issues.redhat.com//browse/CSSRE-2260)

### Pre-checks (if applicable):
- [x] Tested latest changes against a cluster
- [x] Included documentation changes with PR
- [x] If this is a new object that is not intended for the FedRAMP environment (if unsure, please reach out to team FedRAMP), please exclude it with:

    ```yaml
    matchExpressions:
    - key: api.openshift.com/fedramp
      operator: NotIn
      values: ["true"]
    ```

### CODE REVIEW

File: deploy/backplane/cssre/rhoc/02-rhoc-cssre-admins-cluster.SubjectPermission.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+apiVersion: managed.openshift.io/v1alpha1
+kind: SubjectPermission
+metadata:
+  name: backplane-cssre-rhoc-admins-project
+  namespace: openshift-rbac-permissions
+spec:
+  permissions:
+  - allowFirst: true
+    clusterRoleName: backplane-cssre-rhoc-project
+    namespacesAllowedRegex: (^redhat-openshift-connectors$)
```
Comment (mrbarge) [inline deploy/backplane/cssre/rhoc/02-rhoc-cssre-admins-cluster.SubjectPermission.yml:10]: Just checking here - I see some `SubjectPermission` RBAC elsewhere defined in MCC that gets applied to `redhat-openshift-connectors.*` (ie. with the tailing wildcard) and just wanted to check whether that needs to be set here as well.

**Comment by mrbarge** [comment]
@pat-cremin could you please fill out the PR description template?

File: deploy/backplane/cssre/rhoc/02-rhoc-cssre-admins-cluster.SubjectPermission.yml (hunk 1 above)
Comment (pat-cremin) [reply deploy/backplane/cssre/rhoc/02-rhoc-cssre-admins-cluster.SubjectPermission.yml:10]: I know the RHOC team have their own permission set if that is what you are referring to; if so, yes we would need it here as well, as the assumption would be that the Engineering team will possibly be removing these permissions at some point.

**Comment by pat-cremin** [comment]
@mrbarge Updated the template and currently trying to find out the requirements re: FedRAMP.

**Comment by pat-cremin** [comment]
@mrbarge can this be reviewed once more?

File: deploy/backplane/cssre/rhoc/02-rhoc-cssre-admins-cluster.SubjectPermission.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+apiVersion: managed.openshift.io/v1alpha1
+kind: SubjectPermission
+metadata:
+  name: backplane-cssre-rhoc-admins-project
+  namespace: openshift-rbac-permissions
+spec:
+  permissions:
+  - allowFirst: true
+    clusterRoleName: backplane-cssre-rhoc-project
```
Comment (jewzaam) [inline deploy/backplane/cssre/rhoc/02-rhoc-cssre-admins-cluster.SubjectPermission.yml:9]:
```suggestion
    clusterRoleName: backplane-cssre-rhoc-admins-project
```

**Review by jewzaam — CHANGES_REQUESTED** [review]

**Review by jewzaam — CHANGES_REQUESTED** [review]
ClusterRole referenced in SubjectPermission is still wrong, 'admins' is plural in the ClusterRole.

Please update the PR and test this.

File: deploy/backplane/cssre/rhoc/02-rhoc-cssre-admins-cluster.SubjectPermission.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+apiVersion: managed.openshift.io/v1alpha1
+kind: SubjectPermission
+metadata:
+  name: backplane-cssre-rhoc-admins-project
+  namespace: openshift-rbac-permissions
+spec:
+  permissions:
+  - allowFirst: true
+    clusterRoleName: backplane-cssre-rhoc-admin-project
```
Comment (mrbarge) [inline deploy/backplane/cssre/rhoc/02-rhoc-cssre-admins-cluster.SubjectPermission.yml:9]: Thinking that this needs to be `backplane-cssre-rhoc-admins-project` to align with the ClusterRole name defined elsewhere in the PR.

Comment (Nanyte25) [reply deploy/backplane/cssre/rhoc/02-rhoc-cssre-admins-cluster.SubjectPermission.yml:9]: @pat-cremin can you update

File: hack/00-osd-managed-cluster-config-integration.yaml.tmpl [hunk 4]
```diff
@@ -2761,6 +2761,133 @@ objects:
...
+    clusterDeploymentSelector:
+      matchLabels:
+        api.openshift.com/managed: 'true'
+        api.openshift.com/addon-connectors-operator: 'true'
+    resourceApplyMode: Sync
+    resources:
+    - apiVersion: rbac.authorization.k8s.io/v1
+      kind: ClusterRole
+      metadata:
+        name: backplane-cssre-rhoc-admins-project
```
Comment (Nanyte25) [inline hack/00-osd-managed-cluster-config-integration.yaml.tmpl:2782]: @pat-cremin it looks like the `backplane-cssre-rhoc-admins-project`is correct. @mrbarge

Comment (pat-cremin) [reply hack/00-osd-managed-cluster-config-integration.yaml.tmpl]: I had done so, it must have been an older version that had make run against it( i am unable to run make without error). 

Re-pushed and waiting on @Nanyte25 to run make for me.

Comment (Nanyte25) [reply hack/00-osd-managed-cluster-config-integration.yaml.tmpl]: @pat-cremin I have ran the make and push the hack file to you branch

**Review by jewzaam — COMMENTED** [review]
@pat-cremin changes look reasonable, you need to run `make` though to generate stuff.

/approve

**Comment by mrbarge** [comment]
Thanks @pat-cremin 
/lgtm

**Pull request merged**